### ISSUE TITLE
Indexes on Computed Properties cause Invalid Reference Error

### ISSUE THREAD

**Issue by zchenb** [edgedb/edgedb#5548]
EdgeDB 4.0-dev.7528+local (repl 3.1.0-dev)

Reproduction:
1. run `edb cli migration create` on the following schema:

```
module default {
type Person {

    required property name -> str;
    
    property name2 := .name;

    index on (.name2);
}

}
```

The error is:
```
edgedb error: InvalidReferenceError: object type 'default::Person' has no link or property 'name2' (on line 1, column 1)
  Hint: did you mean 'name'?
<stacktraces>
        File "edgedb/edb/edgeql/compiler/setgen.py", line 843, in resolve_ptr_with_intersections
      edb.errors.InvalidReferenceError: object type 'default::Person' has no link or property 'name2'
```

**Comment by zchenb** [comment]
It turns out that this bug only occurs if 
```
using future nonrecursive_access_policies;
``` 
is present in `futures.esdl`.


Removing the future usage will fix this bug.

**Comment by msullivan** [comment]
That is extremely bizarre.

**Comment by zchenb** [comment]
I managed to get it work after commenting out the contents of `futures.esdl`, but that behavior is not stable.

**Comment by zchenb** [comment]
The cause of the issue is that in the migration target schema, the `index` should be a referrer of `name2` property, but it isn't. This causes the delta to be computed incorrectly (out of order), which subsequently caused the wrong ordering of DDL statements that put `CreateIndex` before `CreateProperty` `name2`.

**Comment by zchenb** [comment]
Fixed as of  5.4+bca877e
